eslint-custom-rules/no-useless-hook.js

```diff
@@ -0,0 +1,117 @@
+module.exports = {
+  name: 'no-useless-hook',
+  meta: {
+    docs: {
+      description: "Prevent creating hooks that don't use other hooks",
+      recommended: true,
+    },
+    messages: {
+      noUselessHook:
+        '"{{ name }}" is named like a hook but doesn\'t use any hooks. Rename it without the "use" prefix.',
+    },
+  },
+  create(context) {
+    // Ignore test files
+    const filename = context.getFilename()
+    if (
+      filename.includes('/__tests__/') ||
+      filename.includes('/__mocks__/') ||
+      filename.endsWith('.test.ts') ||
+      filename.endsWith('.test.tsx')
+    ) {
+      return {}
+    }
+
+    const fs = require('fs')
+    const path = require('path')
+    const dir = path.dirname(filename)
+    const baseName = path.basename(filename, path.extname(filename))
+    const isWebFile = filename.includes('.web.')
+
+    const getCorrespondingFile = (baseName) => {
+      const extensions = ['.ts', '.tsx']
+      if (isWebFile) {
+        return extensions.map((ext) => path.join(dir, `${baseName}${ext}`))
+      } else {
+        return extensions.map((ext) => path.join(dir, `${baseName}.web${ext}`))
+      }
+    }
+
+    const correspondingFiles = getCorrespondingFile(baseName)
+
+    const hasCorrespondingFileWithHooks = correspondingFiles.some((file) => {
+      if (!fs.existsSync(file)) return false
+
+      try {
+        const content = fs.readFileSync(file, 'utf8')
+        return content.includes('use') && content.includes('(')
+      } catch (e) {
```

Comment:
pour info, en JS moderne, il n'est plus nécessaire de préciser une variable pour catcher l'erreur si on n'en fait rien

```suggestion
      } catch {
```